src/libs/Log.js

```diff
@@ -33,6 +39,10 @@ const Log = new Logger({
     },
     isDebug: !CONFIG.IS_IN_PRODUCTION,
 });
+info = Log.info;
+timeout = setTimeout(() => info('Flushing logs older than 10 minutes', true, {}, true), 10 * 60 * 1000);
```

Comment:
This is the only thing I could think that overcame this circular dependency. I am happy to do something better, but I don't get what you are suggesting exactly... move it to logger is not possible since logger knows nothing about the API (since one is in App and the other expensify-commons)...

Reply:
Another way to handle this (without messing with the Logger anymore) would be to decorate then re-export the methods from Log and have the timeouts reset when any are used.

```js
function flushAndLog(method) {
    return (...args) => {
        clearTimeout(timeout);
        timeout = setTimeout(() => Log.info('Flushing logs older than 10 minutes', true, {}, true), 10 * 60 * 1000);
        return method(...args);
    }
}

export default {
    info: flushAndLog(Log.info),
    warn: flushAndLog(Log.warn),
    ...
}
```

This way `serverLoggingCallback()` can still remain a dumb callback that just takes the params and logs it - we are just moving the "timer" logic to _before_ we call the methods vs. in the callback. 

Which thinking more on this... isn't that what we want anyway? I thought the `serverLoggingCallback()` is only called when the logs are sent to the server?

Reply:
That's not really what I wanted (or intended to do)... I am unsure why we would care to reset the timer 🤷 
This is working as is, do we really want/need to change it more? What's the current problem?

Reply:
>That's not really what I wanted (or intended to do)...

Ah, I might not have understood this then. Is the intention that we should only "reset the timer" when the log packet arrives to this callback? 

Making the server logging callback log something else will also trigger the callback again eventually and seemed weird to me, but maybe there's no real problem with it.

>This is working as is, do we really want/need to change it more? What's the current problem?

The only _real_ problem I have with this is the reference to `Log.info` that needs to be set up in order for it to work. But another way around that could be to pass the instance to the `serverLoggingCallback` so that we can access it directly like

```js
function serverLoggingCallback(params, logger) {
    logger.info('blah');
    ...
}
```

Reply:
Lines 43 and 27 are also identical which made me think this should be some internal logic of the logger and not something we have to think about in E.cash only. But NAB on that.